services/QuillLMS/app/models/concerns/public_progress_reports.rb

```diff
@@ -54,14 +55,14 @@ def results_by_question(params)
         curr_quest[:question_number] ||= answer["question_number"]
         if answer["attemptNumber"] == 1 || !curr_quest[:instructions]
           direct = answer["directions"] || answer["instructions"] || ""
-          curr_quest[:instructions] ||= direct.gsub(/(<([^>]+)>)/i, "").gsub("()", "").gsub("&nbsp;", "")
+          curr_quest[:instructions] = direct.gsub(/(<([^>]+)>)/i, "").gsub("()", "").gsub("&nbsp;", "")
```

Comment:
Why are we changing from `||=` to just `=` here?

Reply:
Good question.  So what's happening here is that `all_answers`, which is what we're iterating over here, doesn't seem to come back in a sorted order.  So you may get the third attempt first, then the second, then the first.  In the old logic, this resulted in the `:instructions` key being set by attempt 3 which, as far as I can tell, is supposed to be an emergency fallback for cases where we don't have an `attemptNumber == 1` condition.  When this happened, the `instructions` value was set for by a later attempt, which means that it was being set to the value of some attempt's feedback.

This change should result in, at most, one extra set of this value because of the condition on line 56 which will only try to set `:instructions` if it is not set OR this is the first attempt that we're looking at.  So the logic around `||=` is kind of already in the conditional.